### PR TITLE
ref(profiling): Refactor profile processing into its own crate

### PR DESCRIPTION
**Pull request by phacops**
As we're putting more and more code into it, @jan-auer requested we separate our code into its own crate.

### CODE REVIEW

File: relay-profiling/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+use serde::Deserialize;
```
Comment (jjbayer) [inline relay-profiling/src/lib.rs:96]: Should we add some crate-level documentation here? See for example https://github.com/getsentry/relay/blob/d88d327c9f919c4a7c2d9f32cc365bd56c3d3e5b/relay-metrics/src/lib.rs#L1-L93.

**Review by jjbayer — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,7 @@
 - Extract metrics also from trace-sampled transactions. ([#1317](https://github.com/getsentry/relay/pull/1317))
 - Extract metrics from a configurable amount of custom transaction measurements. ([#1324](https://github.com/getsentry/relay/pull/1324))
 - Metrics: Drop transaction tag for high-cardinality sources. ([#1339](https://github.com/getsentry/relay/pull/1339))
+- Refactor profile processing into its own crate. ([#1340](https://github.com/getsentry/relay/pull/1340))
```
Comment (jan-auer) [inline CHANGELOG.md]: Please move this changelog entry, it points to an old release by now.

File: relay-profiling/src/android.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
+use serde::{Deserialize, Serialize};
+
+use crate::utils::deserialize_number_from_string;
+use crate::ProfileError;
+use android_trace_log::AndroidTraceLog;
+use relay_general::protocol::EventId;
```
Comment (jan-auer) [inline relay-profiling/src/android.rs]: Ideally, please sort these imports like that:

```suggestion
use android_trace_log::AndroidTraceLog;

use relay_general::protocol::EventId;

use crate::utils::deserialize_number_from_string;
use crate::ProfileError;
```

This also applies to the other files.

File: relay-profiling/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+publish = false
+
+[dependencies]
+failure = "0.1.8"
+android_trace_log = { version = "0.2.0", features = ["serde"] }
+base64 = "0.10.1"
+bytes = { version = "0.4.12", features = ["serde"] }
+chrono = { version = "0.4.11", features = ["serde"] }
+relay-common = { path = "../relay-common" }
+relay-config = { path = "../relay-config" }
```
Comment (jan-auer) [inline relay-profiling/Cargo.toml]: It looks like the dependencies on `relay-common` and `relay-config` are not actually needed, so they can be removed here.

**Review by jan-auer — APPROVED** [review]
Minor comments below. 

Thanks for the refactor, this indeed makes the code very readable and adds clear separation.

**Pull request merged**